### ISSUE TITLE
Abort Running Requests from Response Interceptor

### ISSUE THREAD

**Issue by gravy-jones-locker** [wkeeling/selenium-wire#558]
Hi, I don't know if this is a suggestion or an issue or an impossibility or an idiocy.

But I have a use-case where I would like to interrupt a request as soon as I have access to a 'Content-Type' response header. Basically I don't want Selenium Wire to download large .pdf files when it navigates to a link which would otherwise prompt that behaviour. The links themselves do not indicate whether or not this would be the case.

I can inspect that header using a simple response interceptor but it seems that running `request.abort()` on the corresponding Request object has no effect - probably because it's already completed? My other idea has been to run `request.head(request.url)` from within a request interceptor - but the site in question only returns CloudFlare headers.

Is this something that can already be done? And if not is it something that can be implemented within the scope of Selenium Wire? Cheers.

**Comment by wkeeling** [comment]
Yes by the time the response interceptor executes, the response has already come back and the file downloaded (as far as Selenium Wire at least). You can't abort the request at that point as you discovered, but you could try setting the body of the response to empty bytes `b''` and the `Content-Length` header to 0. That might make things run a little quicker at the point Selenium Wire hands back to the browser, although I suspect you may have suffered most of the latency by that point anyway.

Running `request.head()` from the interceptor sounds like it may be a better approach if you can avoid CloudFlare. I guess you're using `requests` to make the call? CloudFlare can probably detect that. Have you ensured that all of the headers are transferred from the original GET request to the HEAD request - to ensure that the request from `requests` looks the same?

**Comment by gravy-jones-locker** [comment]
! This is exactly the kind of thing I was looking for. I will give it a try and let you know how it goes. Being able to edit response bodies on the fly before they (or the underlying requests) have finished execution does seem like a cool feature.

**Comment by FloatingMind12** [comment]
Did you solved this issue?
How to get the response headers before the entire request finishes
Using `requests.head(request.url)` also doesn't work in my case because it returns something else
